src/content/config.ts

```diff
@@ -1,5 +1,23 @@
 import { z, defineCollection } from "astro:content"
 
+enum Products {
+  CCIP = "ccip",
+  AUTOMATION = "automation",
+  FUNCTIONS = "functions",
+  VRF = "vrf",
+  FEEDS = "feeds",
+}
+
+export const productsInfo: Record<Products, { name: string; slug: string }> = {
+  ccip: { name: "CCIP", slug: "ccip" },
+  automation: { name: "Automation", slug: "chainlink-automation/introduction" },
+  functions: { name: "Functions", slug: "chainlink-functions" },
+  vrf: { name: "VRF", slug: "vrf/v2/introduction" },
+  feeds: { name: "Data Feeds", slug: "data-feeds" },
+}
+
+const productEnum = z.preprocess((val) => (val as string).toLowerCase(), z.nativeEnum(Products))
```

Comment:
I don't think the preprocessing is necessary, we want to ensure that the frontmatter is consistent and not allow values such as `FuNcTioNs`

Reply:
While it's not strictly necessary if people abide by the exact casing, it makes it easier for people to write frontmatter content (specifically the products property) that is inferred as it's meant to be.